modules/loader-utils/src/worker-utils/create-worker.js

```diff
@@ -9,32 +9,65 @@ export default function createWorker(loader) {
     return;
   }
 
-  self.onmessage = evt => {
+  self.onmessage = async evt => {
```

Comment:
Nit: I'd split this into a generic `createWorker` and a `createLoaderWorker` that adds the extra protocol processing.

The could be named differently (e.g createGenericWorker and createWorker) to not break the existing loader objects that depend on `createWorker`, but separating the basics and the "protocol" like you did on the calling side.

Reply:
There is no breaking change to the worker code. The only addition is a global `parse` function that communicates with the main thread.